runtime/src/main/java/org/corfudb/runtime/view/stream/StreamContext.java

```diff
@@ -0,0 +1,103 @@
+package org.corfudb.runtime.view.stream;
+
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.view.Address;
+
+import java.util.NavigableSet;
+import java.util.TreeSet;
+import java.util.UUID;
+
+
+@ToString
+@Slf4j
+class StreamContext extends AbstractStreamContext {
+
+    /**
+     * A queue of addresses which have already been resolved.
+     */
+    final NavigableSet<Long> resolvedQueue
+            = new TreeSet<>();
+
+    /**
+     * The minimum global address which we have resolved this
+     * stream to.
+     */
+    long minResolution = Address.NON_ADDRESS;
+
+    /**
+     * The maximum global address which we have resolved this
+     * stream to.
+     */
+    long maxResolution = Address.NON_ADDRESS;
+
+    /**
+     * A priority queue of potential addresses to be read from.
+     */
+    final NavigableSet<Long> readQueue
+            = new TreeSet<>();
+
+    /**
+     * List of checkpoint records, if a successful checkpoint has been observed.
+     */
+    final NavigableSet<Long> readCpQueue = new TreeSet<>();
+
+    /**
+     * Info on checkpoint we used for initial stream replay,
+     * other checkpoint-related info & stats.
+     */
+    AbstractQueuedStreamView.StreamCheckpoint checkpoint = new AbstractQueuedStreamView.StreamCheckpoint();
+
+    /**
+     * Create a new stream context with the given ID and maximum address
+     * to read to.
+     *
+     * @param id               The ID of the stream to read from
+     * @param maxGlobalAddress The maximum address for the context.
+     */
+    public StreamContext(UUID id, long maxGlobalAddress) {
+        super(id, maxGlobalAddress);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    void reset() {
```

Comment:
This function would better be synchronized. Since all streamview operations are wrapped in ThreadSafeStreamView, it is also possible to not use synchronization in this function but synchronization should also be removed at seek().